hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -1120,21 +1120,26 @@ protected Map<String, Option<HoodiePendingRollbackInfo>> getPendingRollbackInfos
       HoodieRollbackPlan rollbackPlan;
       try {
         rollbackPlan = RollbackUtils.getRollbackPlan(metaClient, rollbackInstant);
-      } catch (Exception e) {
-        if (rollbackInstant.isRequested()) {
-          LOG.warn("Fetching rollback plan failed for " + rollbackInstant + ", deleting the plan since it's in REQUESTED state", e);
-          try {
-            metaClient.getActiveTimeline().deletePending(rollbackInstant);
-          } catch (HoodieIOException he) {
-            LOG.warn("Cannot delete " + rollbackInstant, he);
-            continue;
+      } catch (IOException ioe) {
+        if (ioe.getMessage().contains("Not an Avro data file")) {
```

Comment:
Relying on an exception message can be problematic as it can change across versions. Is there another way? Might not be.

Reply:
Check my comment here: https://github.com/apache/hudi/pull/5245/files#r932532922.  I don't think we should merge this change.

Reply:
synced w/ Ethan on this. Based on the comment [here](https://github.com/apache/hudi/pull/5245/files#r932532922), its intentional that, if rollback.requested is not accessible for any reason, we will go ahead and regenerate a new rollback plan (and delete the old one). This is the safest option we can think of compared to other alternatives. 
btw, we do this only incase of rollback.requested and not for rollback.inflight. So, we should be good.